server/app/vfg/parser/Predicates_generator.py

```diff
@@ -128,6 +128,98 @@ def get_stages(plan, problem_dic, problem_file, predicates_list):
     return content
 
 
+def new_get_stages(plan, problem_dic, problem_file, predicates_list):
```

Comment:
Why do we need to keep both `get_stages` and `new_get_stages`? It would be preferable to keep only one function. If the new function is needed, then it seems to share too much code with the old function. Furthermore naming a function `new_*` is not a good convention, as it doesn't tell you anything about the function. Can you please review this code? Thanks!